# Importing NY Listings csv

In [7]:
#import libraries, create pathname
import pandas as pd
import numpy as np
import os
import matplotlib.pyplot as plt
import seaborn as sns
import scipy
path=r'/Users/evancarr/Desktop/New York CSVs/NYlistings.csv'

In [11]:
nylist=pd.read_csv(r'/Users/evancarr/Desktop/New York CSVs/NYlistings.csv')

In [20]:
nylist.describe()

id       host_id      latitude     longitude          price  \
count  3.776500e+04  3.776500e+04  37765.000000  37765.000000   23044.000000   
mean   3.653038e+17  1.659684e+08     40.729075    -73.947084     221.074900   
std    4.539908e+17  1.800529e+08      0.056343      0.054458     766.620992   
min    2.595000e+03  1.678000e+03     40.500366    -74.251907       8.000000   
25%    2.055254e+07  1.662776e+07     40.688800    -73.983220      86.000000   
50%    4.826382e+07  8.218953e+07     40.726440    -73.954540     150.000000   
75%    8.276380e+17  3.031569e+08     40.762470    -73.928022     250.000000   
max    1.193086e+18  5.869174e+08     40.911390    -73.713650  100000.000000   

       minimum_nights  number_of_reviews  reviews_per_month  \
count    37765.000000       37765.000000       26014.000000   
mean        29.164597          24.898981           0.905544   
std         30.202538          58.802122           1.860784   
min          1.000000           0.000000           0.010000   
25%         30.000000           0.000000           0.090000   
50%         30.000000           3.000000           0.320000   
75%         30.000000          21.000000           1.130000   
max       1250.000000        1915.000000         103.530000   

       calculated_host_listings_count  availability_365  number_of_reviews_ltm  
count                    37765.000000      37765.000000           37765.000000  
mean                        53.763035        152.411598               3.993168  
std                        161.979723        142.797333              15.762591  
min                          1.000000          0.000000               0.000000  
25%                          1.000000          0.000000               0.000000  
50%                          2.000000        126.000000               0.000000  
75%                          8.000000        301.000000               3.000000  
max                        842.000000        365.000000            1568.000000

## price is missing 14,721 values
## maximum in the price column seems too high (100,000)
## maximum on minimum_nights seems high (1250)
## Maximum number_of_reviews_ltm seems high (1568)

In [22]:
nylist.head()

id                                               name  host_id  \
0  2595                              Skylit Midtown Castle     2845   
1  5136    Spacious Family Friendly Duplex w/ Patio + Yard     7378   
2  6848                   Only 2 stops to Manhattan studio    15991   
3  6872  Uptown Sanctuary w/ Private Bath (Month to Month)    16104   
4  6990                            UES Beautiful Blue Room    16800   

       host_name neighbourhood_group neighbourhood  latitude  longitude  \
0       Jennifer           Manhattan       Midtown  40.75356  -73.98559   
1        Rebecca            Brooklyn   Sunset Park  40.66265  -73.99454   
2  Allen & Irina            Brooklyn  Williamsburg  40.70935  -73.95342   
3            Kae           Manhattan   East Harlem  40.80107  -73.94255   
4            Cyn           Manhattan   East Harlem  40.78778  -73.94759   

         room_type  price  minimum_nights  number_of_reviews last_review  \
0  Entire home/apt    NaN              30                 49  2022-06-21   
1  Entire home/apt  215.0              30                  4  2023-08-20   
2  Entire home/apt   81.0              30                193  2024-05-18   
3     Private room   65.0              30                  1  2022-06-05   
4     Private room   65.0              30                247  2024-03-06   

   reviews_per_month  calculated_host_listings_count  availability_365  \
0               0.28                               3               300   
1               0.03                               1                71   
2               1.05                               1               193   
3               0.04                               2               365   
4               1.38                               1               212   

   number_of_reviews_ltm license  
0                      0     NaN  
1                      1     NaN  
2                      3     NaN  
3                      0     NaN  
4                      2     NaN

# Consistency Checks

In [26]:
#checking for null values
nylist.isnull().sum()

id                                    0
name                                  2
host_id                               0
host_name                             5
neighbourhood_group                   0
neighbourhood                         0
latitude                              0
longitude                             0
room_type                             0
price                             14721
minimum_nights                        0
number_of_reviews                     0
last_review                       11751
reviews_per_month                 11751
calculated_host_listings_count        0
availability_365                      0
number_of_reviews_ltm                 0
license                           32594
dtype: int64

### name and host name may be private/ personal data 
### Must impute average price ($221.75) so as not to skew price data
### last_review and reviews_per_month are the same
### most of license column is blank/ may turn into boolean data

In [35]:
#checking shape and creating data frame without duplicates
nylist.shape

(37765, 18)

In [38]:
nylist_clean= nylist.drop_duplicates()

In [40]:
nylist_clean.shape

(37765, 18)

### No duplicates were found. Will revert back to nylist dataframe

In [45]:
nylist.head()

id                                               name  host_id  \
0  2595                              Skylit Midtown Castle     2845   
1  5136    Spacious Family Friendly Duplex w/ Patio + Yard     7378   
2  6848                   Only 2 stops to Manhattan studio    15991   
3  6872  Uptown Sanctuary w/ Private Bath (Month to Month)    16104   
4  6990                            UES Beautiful Blue Room    16800   

       host_name neighbourhood_group neighbourhood  latitude  longitude  \
0       Jennifer           Manhattan       Midtown  40.75356  -73.98559   
1        Rebecca            Brooklyn   Sunset Park  40.66265  -73.99454   
2  Allen & Irina            Brooklyn  Williamsburg  40.70935  -73.95342   
3            Kae           Manhattan   East Harlem  40.80107  -73.94255   
4            Cyn           Manhattan   East Harlem  40.78778  -73.94759   

         room_type  price  minimum_nights  number_of_reviews last_review  \
0  Entire home/apt    NaN              30                 49  2022-06-21   
1  Entire home/apt  215.0              30                  4  2023-08-20   
2  Entire home/apt   81.0              30                193  2024-05-18   
3     Private room   65.0              30                  1  2022-06-05   
4     Private room   65.0              30                247  2024-03-06   

   reviews_per_month  calculated_host_listings_count  availability_365  \
0               0.28                               3               300   
1               0.03                               1                71   
2               1.05                               1               193   
3               0.04                               2               365   
4               1.38                               1               212   

   number_of_reviews_ltm license  
0                      0     NaN  
1                      1     NaN  
2                      3     NaN  
3                      0     NaN  
4                      2     NaN

In [53]:
nylist.dtypes

id                                  int64
name                               object
host_id                             int64
host_name                          object
neighbourhood_group                object
neighbourhood                      object
latitude                          float64
longitude                         float64
room_type                          object
price                             float64
minimum_nights                      int64
number_of_reviews                   int64
last_review                        object
reviews_per_month                 float64
calculated_host_listings_count      int64
availability_365                    int64
number_of_reviews_ltm               int64
license                            object
dtype: object

## license dtype should be turned to boolean and categorized as yes/no

# testing for mixed type data

In [58]:
for col in nylist.columns.tolist():
  weird = (nylist[[col]].applymap(type) != nylist[[col]].iloc[0].apply(type)).any(axis = 1)
  if len (nylist[weird]) > 0:
    print (col)

name
host_name
last_review
license


/var/folders/z7/kqs18yh15zq9sxssld03yyvh0000gn/T/ipykernel_12778/2529852242.py:2: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.
  weird = (nylist[[col]].applymap(type) != nylist[[col]].iloc[0].apply(type)).any(axis = 1)
/var/folders/z7/kqs18yh15zq9sxssld03yyvh0000gn/T/ipykernel_12778/2529852242.py:2: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.
  weird = (nylist[[col]].applymap(type) != nylist[[col]].iloc[0].apply(type)).any(axis = 1)
/var/folders/z7/kqs18yh15zq9sxssld03yyvh0000gn/T/ipykernel_12778/2529852242.py:2: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.
  weird = (nylist[[col]].applymap(type) != nylist[[col]].iloc[0].apply(type)).any(axis = 1)
/var/folders/z7/kqs18yh15zq9sxssld03yyvh0000gn/T/ipykernel_12778/2529852242.py:2: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.
  weird = (nylist[[col]].applymap(type) != nylist[[col]].iloc[0].app

# Exporting 1st draft of nylistings

In [83]:
nylist.to_csv(os.path.join('/Users/evancarr/Desktop/New York Python Project/prepared data','nylisttemplate.csv'))